Assets/Scripts/CharacterBehaviour.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CharacterBehaviour : MonoBehaviour
+{
+    //MOUSE CONTROL
+    public float lookSpeed, lookXLimit;
+    float rotationX = 0;
+
+    //MOVEMENT PROPERTIES
+    CharacterController characterController;
+    public float walkingSpeed = 5;
+    public float runningSpeed, jumpForce;
+    public bool canMove = true;
+    Vector3 moveDirection = Vector3.zero;
+
+    //PRIVATE ESSENTIALS
+    Camera playerCamera;
+    
+    
+    void Start()
+    {
+        Cursor.lockState = CursorLockMode.Locked;
+
+        characterController = GetComponent<CharacterController>();
+        playerCamera = FindObjectOfType<Camera>().GetComponent<Camera>();
+
+        canMove = true;
+    }
+
+    void Update()
+    {
+        // We are grounded, so recalculate move direction based on axes
+        Vector3 forward = transform.TransformDirection(Vector3.forward);
+        Vector3 right = transform.TransformDirection(Vector3.right);
+        // Press Left Shift to run
+        bool isRunning = Input.GetKey(KeyCode.LeftShift);
+        float curSpeedX = canMove ? (isRunning ? runningSpeed : walkingSpeed) * Input.GetAxis("Vertical") : 0;
```

Comment:
avoid avoid avoid nested ternary operators. i know it's cool to use and we all get to feel clever lmao but it is incredibly hard to understand this line and the one below at a glance

Reply:
Understandable. This was admittedly copy-pasted code as I was tired at the time of the code. At the moment, it works, but if further adjustment is needed down the line, this will definitely be changed.